lib/code_writer.rb

```diff
@@ -12,25 +12,15 @@ def initialize(out)
   }
 
   def write_push_pop(command, segment, index)
-    if command == :C_POP
-      if segment == "temp"
-        final_memory_address = "@#{5+index}"
-      else
-        final_memory_address = <<~EOF
-          @#{index}
-          D=A
-          @#{SEGMENT_TO_SYMBOL_HASH[segment]}
-          A=M+D
-        EOF
-      end
-
+    case command
```

Comment:
Nice! This communicates useful detail up-front about this long conditional: only the value of `command` affects which branch will be taken.